### PR TITLE
Fix page

### LINKED ISSUE #679: search - reset to page 1 when filters change
### From partner feedback
the page number doesn’t reset if you clicked to a later page and then do a new search from that page.

For example, this url works:  https://openaq-staging.org/#/locations?mobility=Mobile

Now scroll to bottom and click the page 2 gets you to this page:  https://openaq-staging.org/#/locations?mobility=Mobile&page=2

From there, select NO2 as the parameter as additional filter, takes you to this page:  https://openaq-staging.org/#/locations?mobility=Mobile&page=2&parameters=7

No locations show up on that page because there are only 6 results with NO2 so page 2 shows up as empty.

**Can you get the page to go back to the beginning when the filters change?**

Page shown when the filters are reset:  https://openaq-staging.org/#/locations?mobility=Mobile&parameters=7

### PR DESCRIPTION
**Pull request by jvntf**
closes #679

### CODE REVIEW

File: app/assets/scripts/components/card/card.js [hunk 1]
```diff
@@ -65,6 +65,7 @@ export default function Card({
   noBodyStyle,
   titleInfo,
 }) {
+  console.log(tags);
```
Comment (AliceR) [inline app/assets/scripts/components/card/card.js:68]:
```suggestion

    console.log(tags);
```

File: app/assets/scripts/views/locations-hub/locations-hub.js [hunk 2]
```diff
@@ -55,7 +55,15 @@ export default function LocationsHub({
     let query = qs.parse(location.search, {
       ignoreQueryPrefix: true,
     });
-    setPage(() => getPage(query));
+    setPage(prev => {
+      const cur = getPage(query);
+      if (prev === cur) {
+        //This means that only filtes have updated
```
Comment (AliceR) [inline app/assets/scripts/views/locations-hub/locations-hub.js:61]: 👍 thanks for the explanation! Valuable comment

File: cypress/integration/locations-hub.spec.js [hunk 3]
```diff
@@ -90,4 +90,22 @@ describe('The Locations Hub', () => {
     cy.get('[data-cy=location-card-detail-label]').contains('Parameters');
     cy.get('[data-cy=location-card-detail-label]').contains('Source');
   });
+
+  it('Pagination is reset on filter change', () => {
```
Comment (AliceR) [inline cypress/integration/locations-hub.spec.js:94]: 🎉 ❤️ 🥳

**Review by rwegener2 — APPROVED** [review]
🎉

**Review by AliceR — CHANGES_REQUESTED** [review]
I ran into a loop when trying to filter for ow cost sensors. I'll get you more info soon

**Comment by jvntf** [comment]
@AliceR loop fixed

**Review by necoline — APPROVED** [review]

**Comment by AliceR** [comment]
But the page click does not seem to work? It always resets me back to page 1 when I click the pagination button

**Comment by AliceR** [comment]
@jvntf when there is a space in the filter parameter name (`Low-Cost Sensor`), the browser updates to url to `Low-Cost%20Sensor` which triggers the effect while prev and cur page are the same, hence it goes to page 1 again 🙈

**Review by AliceR — APPROVED** [review]

**Review by necoline — APPROVED** [review]

**Pull request merged**